src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -15,242 +14,273 @@
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Simplification;
 
-namespace Microsoft.CodeAnalysis.ConvertToInterpolatedString
+namespace Microsoft.CodeAnalysis.ConvertToInterpolatedString;
+
+/// <summary>
+/// Code refactoring that converts expressions of the form:  a + b + " str " + d + e
+/// into:
+///     $"{a + b} str {d}{e}".
+/// </summary>
+internal abstract class AbstractConvertConcatenationToInterpolatedStringRefactoringProvider<TExpressionSyntax> : CodeRefactoringProvider
+    where TExpressionSyntax : SyntaxNode
 {
-    /// <summary>
-    /// Code refactoring that converts expressions of the form:  a + b + " str " + d + e
-    /// into:
-    ///     $"{a + b} str {d}{e}".
-    /// </summary>
-    internal abstract class AbstractConvertConcatenationToInterpolatedStringRefactoringProvider<TExpressionSyntax> : CodeRefactoringProvider
-        where TExpressionSyntax : SyntaxNode
+    protected abstract bool SupportsInterpolatedStringHandler(Compilation compilation);
+
+    protected abstract string GetTextWithoutQuotes(string text, bool isVerbatimStringLiteral, bool isCharacterLiteral);
+
+    public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
     {
-        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
-        {
-            var (document, textSpan, cancellationToken) = context;
-            var possibleExpressions = await context.GetRelevantNodesAsync<TExpressionSyntax>().ConfigureAwait(false);
+        var (document, textSpan, cancellationToken) = context;
+        var possibleExpressions = await context.GetRelevantNodesAsync<TExpressionSyntax>().ConfigureAwait(false);
 
-            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var generator = SyntaxGenerator.GetGenerator(document);
```

Comment:
>var generator = SyntaxGenerator.GetGenerator(document); [](http://example.com/codeflow?start=8&length=55)

why did this move out to a larger scope?

Reply:
Just preferred it that way.

Reply:
OK, in taking a look, there's not much cost associated with doing that, so whatever looks best is the way to go.